Adds CLI to make downloading species data easier

ndukler opened the pull request:
Added CLI to maintenance script for downloading species data

jeromekelleher commented on update_ensembl_data.py:
```diff
@@ -29,7 +41,7 @@ def write(self, path):
         # TODO we can remove this once we've moved to using pre-commit and
         # Black, as these will be autoformatted.
-        subprocess.check_call(["black", "-q", path])
+        # subprocess.check_call(["black", "-q", path])
```
I think we should keep the ``black`` call here - otherwise the code we're auto inserting will be horrible.

jeromekelleher replied:
Maybe add a comment here for developers for now saying "use ``pip install black`` if this command fails".

ndukler replied:
Ok, I was having trouble with it but once I'm done with local testing I'll un-comment it.

jeromekelleher commented on update_ensembl_data.py:
```diff
@@ -11,7 +11,19 @@
 
+# Parser that controls which species should be updated
+parser = argparse.ArgumentParser(description='Query ensembl API for chromosome lengths')
+parser.add_argument('--list-species', dest='list_species', action='store_true',
```
I think we can simplify this a bit - the ``--list`` isn't really needed since we can get the list of species from a bunch of other places. What about if we had the default of doing what it's currently doing (all species), so if we call
```
python update_ensembl_data.py
```
it'll run through all species and if we call 
```
python update_ensembl_data.py HomSap BosTau
```
it'll just do those two. Basically, it's just a case of putting one positional argument, rather than two mutually exclusive mandatory "options".

ndukler replied:
And the `--list` bit is just so that someone doesn't have to go hunting for a way to list all the registered species. I'm particularly thinking of my current case where a new developer is submitting a species but is not familiar with `stdpopsim` generally.

jeromekelleher replied:
Well, the update_all is harmless as it'll just over-write identical data:
```
$ python3 update_ensembl_data.py 
Using Ensembl release 101
Writing genome data for AraTha arabidopsis_thaliana
Writing genome data for CanFam canis_familiaris
Writing genome data for DroMel drosophila_melanogaster
... [Ctrl-c'd out of boredom]
$ git diff
diff --git a/stdpopsim/catalog/ensembl_info.py b/stdpopsim/catalog/ensembl_info.py
index 4800dda..1d96ff5 100644
--- a/stdpopsim/catalog/ensembl_info.py
+++ b/stdpopsim/catalog/ensembl_info.py
@@ -1,2 +1,2 @@
 # File autogenerated from Ensembl REST API. Do not edit.
-release = 100
+release = 101
```

So, we updated 3 species, but there's no diff in git (other than we seem to have updated to Ensembl release 101).

Having mandatory options (you must specify either --species or --update-all) gives me hives.

Fair enough having --list is handy, so let's keep that.

I have strong opinions about CLI structure I'm afraid @ndukler :wink:

ndukler replied:
I'll change it. We can always catch a surprise unintentional update at the review stage.

jeromekelleher reviewed (commented):
Looks good, thanks @ndukler. I think a small change to the UI would simplify it a lot, but it's definitely a really useful addition to the utility.

jeromekelleher commented on update_ensembl_data.py:
```diff
@@ -41,17 +53,31 @@ def write_genome_data(self, species):
+    if args.list_species:
+        for species in stdpopsim.all_species():
+            print(species.id)
+        exit(0)
```
Does this work? I think you need to call ``sys.exit(0)``, right?

Rather than yanking out of the interpreter directly here, it's a bit more pleasant to return here, and let control flow continue normally.

ndukler commented:
@jeromekelleher Think I fixed the hairball by using ensembl_ids instead and generating the six character ids directly from those. Also added some error checking to make it less likely an invalid species name slips through to the step where ensembl is queried.

ndukler commented:
Just going to update the documentation and then we should be good to go

ndukler commented:
Ok, just squashed stuff. Pending the checks I think that this is good to go @jeromekelleher

jeromekelleher commented:
Looks good, thanks @ndukler
